controllers/secretgenerator/secretgenerator_controller.go

```diff
@@ -132,54 +132,54 @@ func (r *SecretGeneratorReconciler) Reconcile(ctx context.Context, request ctrl.
-				return ctrl.Result{}, err
-			}
+	getErr := r.Client.Get(ctx, generatedSecretKey, generatedSecret)
+	if getErr != nil && k8serr.IsNotFound(getErr) {
+		// Generate secret random value
+		log.Info("Generating a random value for a secret in a namespace",
+			"secret", generatedSecret.Name, "namespace", generatedSecret.Namespace)
 
-			generatedSecret.StringData = map[string]string{
-				secret.Name: secret.Value,
-			}
+		secret, err := NewSecretFrom(foundSecret.GetAnnotations())
+		if err != nil {
+			log.Error(err, "error creating secret %s in %s", generatedSecret.Name, generatedSecret.Namespace)
+			return ctrl.Result{}, err
+		}
+
+		generatedSecret.StringData = map[string]string{
+			secret.Name: secret.Value,
+		}
+
+		err = r.Client.Create(ctx, generatedSecret)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
 
-			err = r.Client.Create(ctx, generatedSecret)
+		log.Info("Done generating secret in namespace",
+			"secret", generatedSecret.Name, "namespace", generatedSecret.Namespace)
+		// check if annotation oauth-client-route exists
+		if secret.OAuthClientRoute != "" {
+			// Get OauthClient Route
+			oauthClientRoute, err := r.getRoute(ctx, secret.OAuthClientRoute, request.Namespace)
 			if err != nil {
+				log.Error(err, "Unable to retrieve route from OAuthClient", "route-name", secret.OAuthClientRoute)
 				return ctrl.Result{}, err
 			}
-			log.Info("Done generating secret in namespace",
-				"secret", generatedSecret.Name, "namespace", generatedSecret.Namespace)
-			// check if annotation oauth-client-route exists
-			if secret.OAuthClientRoute != "" {
-				// Get OauthClient Route
-				oauthClientRoute, err := r.getRoute(ctx, secret.OAuthClientRoute, request.Namespace)
-				if err != nil {
-					log.Error(err, "Unable to retrieve route from OAuthClient", "route-name", secret.OAuthClientRoute)
-					return ctrl.Result{}, err
-				}
-				// Generate OAuthClient for the generated secret
-				log.Info("Generating an OAuthClient CR for route", "route-name", oauthClientRoute.Name)
-				err = r.createOAuthClient(ctx, foundSecret.Name, secret.Value, oauthClientRoute.Spec.Host)
-				if err != nil {
-					log.Error(err, "error creating oauth client resource. Recreate the Secret", "secret-name",
-						foundSecret.Name)
-
-					return ctrl.Result{}, err
-				}
+			// Generate OAuthClient for the generated secret
+			log.Info("Generating an OAuthClient CR for route", "route-name", oauthClientRoute.Name)
+			err = r.createOAuthClient(ctx, foundSecret.Name, secret.Value, oauthClientRoute.Spec.Host)
+			if err != nil {
+				log.Error(err, "error creating oauth client resource. Recreate the Secret", "secret-name",
+					foundSecret.Name)
+
+				return ctrl.Result{}, err
 			}
-		} else {
-			return ctrl.Result{}, err
 		}
+
+		return ctrl.Result{}, nil
 	}
 
 	// Don't requeue if secret is created successfully
-	return ctrl.Result{}, err
```

Comment:
at this point, the `err` was set to the `NotFound` error before the rename because of variable name shadowing